docs/input.md

```diff
@@ -90,20 +90,199 @@ def setup(): # Only drawn once, so just using setup()
 
 which results in:
 
-![Write Name](img/write_name.png)
+![Write Name](img/write_name.png){: loading=lazy }
 
 
-*Ask the user for a radius. We need to convert the string to an integer, we can do this using int(). Draw a circle of that radius at (100, 100):*
+*Ask the user for a diameter. We need to convert the string to an integer, we can do this using int(). Draw a circle of that diameter at (100, 100):*
 
-```python
+```python hl_lines="5"
 %%ignite
 
 def setup(): # Only drawn once, so just using setup()
     size(200, 200)
-    radius = int(input("What radius should the circle be?: ")) # cast the string to an integer
-    fill_circle(100, 100 radius)
+    diameter = int(input("What diameter should the circle be?: ")) # cast the string to an integer
+    fill_circle(100, 100 diameter)
 ```
 
 which results in:
 
-![Ask circle radius](img/circle_radius.png)
+![Ask circle diameter](img/circle_radius.png){: loading=lazy }
+
+### Check for key presses
+
+Spark features multiple ways to check for user key presses:
+
+| Name                            | Type                  | Description                                                  |
+|---------------------------------|-----------------------|--------------------------------------------------------------|
+| [key](#key)                     | Variable              | A string of what key was last pressed                        |
+| [keys_held()](#keys_held)       | Function              | A function that returns True if the provided key is held     |
+| [key_pressed()](#key_pressed)   | User defined function | A definable function that activates when any key is pressed  | 
+| [key_released()](#key_released) | User defined function | A definable function that activates when any key is released |
+| [key_repeated()](#key_repeated) | User defined function | A definable function that activates when any key is held     |
+
+#### key
+
+A string that is the last key pressed
+
+##### Notes
+- This variable **does not clear**, meaning if a key is pressed it will remain the value until a **new key** is pressed
+- This variable will be the uppercase letter if ++shift++ + a letter is pressed i.e. ++shift++ + ++a++ would be `#!python key == "A"`
+- This variable will be a symbol if ++shift++ + a number is pressed i.e. ++shift++ + ++3++ would be `#!python key == "$"`
+
+
+##### Special Keys
+
+| Key           | Value                          |
+|---------------|--------------------------------|
+| ++up++        | `#!python key == "ArrowUp"`    |
+| ++down++      | `#!python key == "ArrowDown"`  |
+| ++left++      | `#!python key == "ArrowLeft"`  |
+| ++right++     | `#!python key == "ArrowRight"` |
+| ++tab++       | `#!python key == "Tab"`        |
+| ++ctrl++      | `#!python key == "Control"`    |
+| ++alt++       | `#!python key == "Alt"`        |
+| ++esc++       | `#!python key == "Escape"`     |
+| ++win++       | `#!python key == "Meta"`       |
```

Comment:
May also want to mention the command key, which I believe also shows up as `"Meta"`? I can't confirm, I haven't had a Mac in years

Reply:
Yeah I wasn't sure on this, do we know anyone who has a mac on the team?

Reply:
I actually might be able to test on a mac tomorrow, but overall it's probably not a big deal either way, mac users are probably sufficiently used to windows key <=> command

Reply:
Okay, got busy yesterday, I'll do merges later today. I did, however, just test on a Mac. Command is Meta, and option is Alt, so if we wanted to include those keys that's how they'd associate. I don't know if the md library you're using includes the mac keys, if it doesn't I don't think it's a big deal.

Reply:
Yeah, it does. I can update the table

Reply:
Table updated in latest commit to include option and command